motionplan/tpspace/ptg.go

```diff
@@ -0,0 +1,79 @@
+//go:build !windows
+
+// Package tpspace defines an assortment of precomputable trajectories which can be used to plan nonholonomic 2d motion
+package tpspace
+
+import (
+	"math"
+
+	"github.com/golang/geo/r3"
+
+	"go.viam.com/rdk/spatialmath"
+)
+
+// PTG is a Parameterized Trajectory Generator, which defines how to map back and forth from cartesian space to TP (alpha, d)
```

Comment:
[nit]: for those who are seeing this for the first time can we specify that TP-space uses polar coordinates